packages/oaf/src/commands/_player.ts

```diff
@@ -0,0 +1,60 @@
+import type { Prisma } from "@prisma/client";
+
+import { prisma } from "../clients/database.js";
+import { kolClient } from "../clients/kol.js";
+
+export function validPlayerIdentifier(identifier: string) {
+  // If a player id: a number!
+  // If a username: 3 to 30 alphanumeric characters, starting with alpha, may contain underscores or spaces
+  return /^([a-zA-Z][a-zA-Z0-9_ ]{2,29})|[0-9]+$/.test(identifier);
+}
+
+export async function findPlayer(where: Prisma.PlayerWhereInput) {
+  const player = await prisma.player.findFirst({
+    where,
+    include: { greenbox: { orderBy: { id: "desc" }, take: 1 } },
+  });
+  if (!player) return null;
+  return { ...player, greenbox: player.greenbox.at(0) ?? null };
+}
+
+export async function identifyPlayer(input: string) {
+  // If the input is a Discord mention, we'll try to set a player identifier there. Otherwise this just gets set to
+  // the same value as input.
+  let playerIdentifier;
+
+  // Whatever happens we'll try to ascertain whether this is a known player. Because we either do so at the start or
+  // at the end, declare a null variable here.
+  let knownPlayer: Awaited<ReturnType<typeof findPlayer>> = null;
```

Comment:
is it worth naming this type, or at least naming `ReturnType<typeof findPlayer>`